internal/venice-client-common/src/main/java/com/linkedin/venice/compression/ZstdWithDictCompressor.java

```diff
@@ -64,13 +62,12 @@ public ByteBuffer decompress(byte[] data, int offset, int length) throws IOExcep
 
   @Override
   public InputStream decompress(InputStream inputStream) throws IOException {
-    return new ZstdInputStream(inputStream).setDict(this.decompressorDictionary);
+    return new ZstdInputStream(inputStream).setDict(this.dictionary);
```

Comment:
The latest version of this lib supports optionally passing a `BufferPool` in this constructor. Would be interesting to experiment with this (not in this PR though).

https://javadoc.io/doc/com.github.luben/zstd-jni/latest/com/github/luben/zstd/ZstdInputStream.html